pkg/server/server.go

```diff
@@ -18,40 +18,28 @@ import (
 )
 
 func parseInputFromRequest(r *http.Request) (file.File, error) {
-	if r.Header.Get("Content-Type") == "application/json" {
-		var body map[string]interface{}
 
-		decoder := json.NewDecoder(r.Body)
-		if err := decoder.Decode(&body); err != nil {
-			return nil, err
-		}
-		defer r.Body.Close()
+	src, _, err := r.FormFile("file")
+	if err != nil {
 
-		str, err := json.Marshal(body)
+		buf, err := ioutil.ReadAll(r.Body)
 		if err != nil {
-			return nil, err
+			return nil, errors.New("unable to read request body")
 		}
 
-		mf, err := file.CreateFile(str)
+		defer r.Body.Close()
+
+		mf, err := file.NewFileFromReader(bytes.NewReader(buf))
```

Comment:
Do we need to consume the entire file just to convert it back into a reader? Streaming large files is much more efficient than reading it all into memory.

Reply:
Above comment, we don't avoid seek function in NewFileFromReader
Unfortunately, request body don't support io.(closer) interface
I am trying to create new reader that support io.(closer) in this code